angular/projects/admin/src/app/services/document.service.ts

```diff
@@ -98,9 +100,19 @@ export class DocumentService {
       .valueChanges();
   }
 
-  query(documentTypeId: string, status: string, queryOpts: DocumentTypeQueryOptions = {}): Observable<Document[]> {
+  query(
+    documentTypeId: string, status: string, queryOpts: DocumentTypeQueryOptions = {}, nextPage?: Boolean, pageIndex?: number
+  ): Observable<Document[]> {
     console.log(`[DocumentService:getDocumentTypeFields] ${documentTypeId}, query=${JSON.stringify(queryOpts)}`);
 
+    console.log('[Next page] ', nextPage);
+    console.log('[page index] ', pageIndex);
+
+    if (pageIndex === 0) {
+      this.firstDoc = [];
+      this.lastDoc = '';
```

Comment:
Is `firstDoc` and `lastDoc` representing different things? Seems strange that name is very similar but one is an array and the other a string.